exchanges/okx/okx_websocket.go

```diff
@@ -1227,34 +1259,46 @@ func (ok *Okx) WSPlaceOrder(arg *PlaceOrderRequestParam) (*PlaceOrderResponse, e
 	input := WsPlaceOrderInput{
 		ID:        randomID,
 		Arguments: []PlaceOrderRequestParam{*arg},
-		Operation: "batch-orders",
+		Operation: okxOpOrder,
 	}
-	respData, err := ok.Websocket.Conn.SendMessageReturnResponse(randomID, input)
+	err = ok.Websocket.AuthConn.SendJSONMessage(input)
 	if err != nil {
 		return nil, err
 	}
-	var placeOrderResponse WSPlaceOrderResponse
-	err = json.Unmarshal(respData, &placeOrderResponse)
-	if err != nil {
-		return nil, err
-	}
-	if !(placeOrderResponse.Code == "0" ||
-		placeOrderResponse.Code == "2") {
-		if placeOrderResponse.Msg == "" {
-			return nil, errNoValidResponseFromServer
-		}
-		return nil, errors.New(placeOrderResponse.Msg)
-	} else if len(placeOrderResponse.Data) == 0 {
-		if placeOrderResponse.Msg == "" {
-			return nil, errNoValidResponseFromServer
+	timer := time.NewTimer(ok.WebsocketResponseMaxLimit)
+	ok.wsMutext.Lock()
+	defer ok.wsMutext.Unlock()
+	for {
+		select {
+		case data := <-ok.wsResponse:
```

Comment:
In that case, we might need a ticker loop to clean the data depending on the timeout.
I thought it would be okay If we use synchronized access for WebSocket requests.